viadot/sources/s3.py

```diff
@@ -24,14 +25,20 @@ def __init__(
         aws_secret_access_key: str = None,
     ):
         if profile_name:
-            self.session = boto3.session.Session(profile_name=profile_name)
+            self.wr_session = boto3.session.Session(profile_name=profile_name)
+            self.s3fs_session = s3fs.S3FileSystem(profile=profile_name)
         elif aws_access_key_id and aws_secret_access_key:
-            self.session = boto3.session.Session(
+            self.wr_session = boto3.session.Session(
                 aws_access_key_id=aws_access_key_id,
                 aws_secret_access_key=aws_secret_access_key,
             )
+            self.s3fs_session = s3fs.S3FileSystem(
```

Comment:
`s3fs.S3FileSystem` is not a session, it only knows how to create one. 

In fact, we should have the same here -- an object should not hold a session as its attribute as then the object is non-serializable and this can cause some issues eg. with Prefect. 

The way to work around this is to use the `@property` decorator implemented in a "singleton" pattern (basically to not create a new session on each access to `self.session`). See https://github.com/dyvenia/viadot/blob/main/viadot/sources/base.py#L195

Reply:
[♻️ moved session/connection creation under @property](https://github.com/dyvenia/viadot/pull/579/commits/62312b60a6de14658c28a17345b222b73fa6a339)

Reply:
[♻️ improved session creation](https://github.com/dyvenia/viadot/pull/579/commits/81808669ab462239aec74e488711b5fb66a256d3)